packages/zql/src/query/query.ts

```diff
@@ -108,6 +109,10 @@ export interface Query<
   TTable extends keyof TSchema['tables'] & string,
   TReturn = PullRow<TTable, TSchema>,
 > {
+  readonly format: Format;
+  hash(): string;
+  updateTTL(ttl: TTL): void;
```

Comment:
This one is the most unfortunate one. My gut feeling is that we should hide these from the common usage...

One suggestion is to have:

```ts
interface Query {
  delegate?: QueryDelegate
}
```

but this is not optimal either.

or we can do 

```ts
export getQueryDelegate(q: Query): QueryDelegate | undefined;
```